src/main/java/core/basesyntax/service/impl/FileReaderCsvImpl.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.Reader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderCsvImpl implements Reader {
+    public static final String CSV_TYPE = ".csv";
+    public static final String FIRST_LINE_FORMAT = "type,fruit,quantity";
+    public static final int MIN_LINES_NUMBER = 2;
+    private static final int COLUMN_NAMES_INDEX = 0;
+    private static final int FIRST_LINE_INDEX = 0;
+
+    @Override
+    public List<String> read(String fileName) {
+        List<String> transactions = new ArrayList<>();
+        if (fileName == null) {
+            throw new NullPointerException("File path is null");
```

Comment:
that's very weird, we don't throw NPE manually and with specific messages, replaces with IllegalArgument in the main code